temporalio/activity.py

```diff
@@ -135,6 +138,41 @@ def _logger_details(self) -> Mapping[str, Any]:
 _current_context: contextvars.ContextVar[_Context] = contextvars.ContextVar("activity")
 
 
+@dataclass
+class _ActivityCancellationDetailsHolder:
+    _details: Optional[ActivityCancellationDetails] = None
+
+    def set_details(self, details: ActivityCancellationDetails) -> None:
+        self._details = details
+
+    @property
+    def details(self) -> Optional[ActivityCancellationDetails]:
+        return self._details
```

Comment:
```suggestion
    details: Optional[ActivityCancellationDetails] = None
```

This is a private dataclass (prefixed with `_`), can just use the field as normal, no need for getter/setter